saw-central/src/SAWCentral/Crucible/MIR/TypeShape.hs

```diff
@@ -93,7 +98,7 @@ data TypeShape (tp :: CrucibleType) where
     UnitShape :: M.Ty -> TypeShape UnitType
     PrimShape :: M.Ty -> BaseTypeRepr btp -> TypeShape (BaseToType btp)
     TupleShape :: M.Ty -> [AgElemShape] -> TypeShape MirAggregateType
-    ArrayShape :: M.Ty -> M.Ty -> TypeShape tp -> TypeShape (MirVectorType tp)
+    ArrayShape :: M.Ty -> M.Ty -> Word -> TypeShape tp -> Word -> TypeShape MirAggregateType
```

Comment:
Please add some Haddock comments to explain the difference between the two `Word` fields.